core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelectorFactory.java

```diff
@@ -98,25 +63,13 @@ public UniformNodeSelectorFactory(
         checkArgument(maxSplitsPerNode >= maxPendingSplitsPerTask, "maxSplitsPerNode must be > maxPendingSplitsPerTask");
         this.maxSplitsWeightPerNode = SplitWeight.rawValueForStandardSplitCount(maxSplitsPerNode);
         this.maxPendingSplitsWeightPerTask = SplitWeight.rawValueForStandardSplitCount(maxPendingSplitsPerTask);
-        this.nodeMapMemoizationDuration = nodeMapMemoizationDuration;
     }
 
     @Override
     public NodeSelector createNodeSelector(Session session, Optional<CatalogName> catalogName)
     {
         requireNonNull(catalogName, "catalogName is null");
-
-        // this supplier is thread-safe. TODO: this logic should probably move to the scheduler since the choice of which node to run in should be
-        // done as close to when the split is about to be scheduled
-        Supplier<NodeMap> nodeMap;
```

Comment:
This commit removes the extra nodeMap cache entirely. The nodeMap is already cached indirectly in `DiscoveryNodeManager`. 

@dain added the nodeMap cache in this PR: https://github.com/trinodb/trino/pull/1049

If people have tips on benchmarking/validating that removing the cache does not cause a performance regression it would be helpful. Naively can run CI several times and compare wallclock. 

If we do find there is a performance regression, can change `DiscoveryNodeManager` to use read/write locks instead of `synchronized`